### PR TITLE
Adding SynthID Text blog post

### PR DESCRIPTION
**Pull request by RyanMullins**
Congratulations! You've made it this far! Once merged, the article will appear at https://huggingface.co/blog. Official articles 
require additional reviews. Alternatively, you can write a community article following the process [here](https://huggingface.co/blog-explorers).

## Preparing the Article

You're not quite done yet, though. Please make sure to follow this process (as documented [here](https://github.com/huggingface/blog/tree/main?tab=readme-ov-file#how-to-write-an-article-)):

- [x] Add an entry to [_blog.yml](https://github.com/huggingface/blog/blob/main/_blog.yml).
- [x] Add a thumbnail. There are no requirements here, but there is a [template](https://github.com/huggingface/blog?tab=readme-ov-file#how-to-get-a-nice-responsive-thumbnail) if it's helpful.
- [x] Check you use a short title and blog path.
- [x] Upload any additional assets (such as images) to the Documentation Images [repo](https://huggingface.co/datasets/huggingface/documentation-images/tree/main/blog). This is to reduce bloat in the GitHub base repo when cloning and pulling. Try to have small images to avoid a slow or expensive user experience.
- [x] Add metadata (such as authors) to your `md` file. You can also specify `guest` or `org` for the authors.
- [x] Ensure the publication date is correct.
- [x] Preview the content. A quick way is to paste the markdown content in https://huggingface.co/new-blog. **Do not click publish**, this is just a way to do an early check.

Here is an example of a complete PR: https://github.com/huggingface/blog/pull/2382

## Getting a Review

Please make sure to get a review from someone on your team or a co-author.
Once this is done and once all the steps above are completed, you should be able to merge.
There is no need for additional reviews if you and your co-authors are happy and meet all of the above.

Feel free to add @osanseviero or @pcuenca as reviewers if you want a final check. They'll be biased toward light reviews
(e.g., check for proper metadata) rather than content reviews unless explicitly asked.

### CODE REVIEW

**Review by sumedhghaisas2 — APPROVED** [review]

File: synthid-text.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+## Configuring a watermark
+
+Watermarks are [configured using a dataclass](https://github.com/huggingface/transformers/blob/v4.46.0/src/transformers/generation/configuration_utils.py) that parameterizes the _g_\-function and how it is applied in the tournament sampling process. Each model you use should have its own watermarking configuration that **_should be stored securely and privately_**, otherwise your watermark may be replicable by others.
+
+You must define two parameters in every watermarking configuration:
+
+- The `keys` parameter is a list of unique, random integers that are used to compute _g_-function scores across the model's vocabulary. The length of this list determines how many layers of watermarking are applied.
```
Comment (SunMarc) [inline synthid-text.md:38]: Could you explain a bit how to best set this arg. I feel like the users won't understand which random number to put as well as how many elements this list should have. What is the effect of having more layers of watermarking. Maybe we can even make this argument optional for the user or replace that with `num_layers` to set instead.

File: synthid-text.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    org: Google DeepMind
+    guest: true
+  - user: sdathath
+    org: Google DeepMind
+    guest: true
+  - user: RyanMullins
+    org: Google DeepMind
+    guest: true
+  - user: joaogante
+    org: Hugging Face
```
Comment (gante) [inline synthid-text.md:15]:
```suggestion
```
Hugging Face is a special case that doesn't need to be specified :) (legacy: all blog posts were from Hugging Face)

File: synthid-text.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  - user: marcsun13
+    org: Hugging Face
```
Comment (gante) [inline synthid-text.md:17]:
```suggestion
```

**Review by SunMarc — APPROVED** [review]
Thanks for this nice blogpost ! Left a few minor comments !

File: synthid-text.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  - user: RaushanTurganbay
+    org: Hugging Face
```
Comment (gante) [inline synthid-text.md:19]:
```suggestion
```

File: synthid-text.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+You can further configure the watermark based on your performance needs. See the [`SynthIDTextWatermarkingConfig` class](https://github.com/huggingface/transformers/tree/v4.46.0/src/transformers/generation/configuration_utils.py#L1461-L1478) for more information. The [research paper](https://www.nature.com/articles/s41586-024-08025-4) includes additional analyses of how specific configuration values affect performance.
+
+## Applying a watermark
+
+Applying a watermark is a straightforward change to your existing generation calls. Once you define your configuration, pass a `SynthIDTextWatermarkingConfig` object as the `watermarking\_config=` parameter to `model.generate()` and all generated text will carry the watermark. Check out the [SynthID Text Space](https://huggingface.co/spaces/google/synthid-text) for an interactive example of watermark application, and see if you can tell.
+
+```py
+from transformers import AutoModelForCausalLM, AutoTokenizer
+from transformers.generation import SynthIDTextWatermarkingConfig
```
Comment (gante) [inline synthid-text.md:50]:
```suggestion
from transformers import AutoModelForCausalLM, AutoTokenizer, SynthIDTextWatermarkingConfig
```

Comment (gante) [reply synthid-text.md:50]: I'm adding it as a top-level import :)

**Review by gante — APPROVED** [review]
A few nits, but LGTM 👍

File: synthid-text.md (hunk 1 above)
Comment (RyanMullins) [reply synthid-text.md:38]: Re-wroded this to provide a default recommendation.

For now we want to keep the API as it is. We may change this in the future as we expand the implementation across other frameworks.

**Pull request merged**